integration/analytics_test.go

```diff
@@ -0,0 +1,174 @@
+	err := af.mss.TearDown()
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	af.tempDir.TearDown()
+	af.k8sFixture.TearDown()
+}
+
+type envVarValue struct {
+	name  string
+	isSet bool
+	val   string
+}
+
+func saveEnvVar(name string) envVarValue {
+	val, isSet := os.LookupEnv(name)
+	return envVarValue{
+		name:  name,
+		isSet: isSet,
+		val:   val,
+	}
+}
+
+func restoreEnvVar(v envVarValue) error {
+	if !v.isSet {
+		return os.Unsetenv(v.name)
+	} else {
+		return os.Setenv(v.name, v.val)
+	}
+}
+
+func (af *analyticsFixture) SetOpt(opt analytics.Opt) {
+	v := saveEnvVar(WindmillDirEnvVarName)
+	err := os.Setenv(WindmillDirEnvVarName, af.tempDir.Path())
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	err = analytics.SetOpt(opt)
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	err = restoreEnvVar(v)
+	if err != nil {
+		af.t.Fatal(err)
+	}
+}
+
+func TestOptedIn(t *testing.T) {
+	f := newAnalyticsFixture(t)
+	defer f.TearDown()
+
+	f.SetOpt(analytics.OptIn)
+
+	f.TiltUp("oneup")
+
+	// ForwardPort will fail if all the pods are not ready.
+	//
+	// We can't use the normal Tilt-managed forwards here because
+	// Tilt doesn't setup forwards when --watch=false.
+	ctx, cancel := context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	f.WaitForAllPodsReady(ctx, "app=oneup")
+
+	var observedEventNames []string
+	for _, c := range f.mss.ma.Counts {
+		observedEventNames = append(observedEventNames, c.Name)
+	}
+
+	// just check that a couple metrics were successfully reported rather than asserting an exhaustive list
+	// the goal is to ensure that analytics is working in general, not to test which specific metrics are reported
+	// and we don't want to have to update this every time we change which metrics we report
+	assert.Contains(t, observedEventNames, "tilt.cmd.up")
+	assert.Contains(t, observedEventNames, "tilt.tiltfile.loaded")
+}
+
+func TestOptedOut(t *testing.T) {
+	f := newAnalyticsFixture(t)
+	defer f.TearDown()
+
+	f.SetOpt(analytics.OptOut)
+
+	f.TiltUp("oneup")
+
+	// ForwardPort will fail if all the pods are not ready.
+	//
+	// We can't use the normal Tilt-managed forwards here because
+	// Tilt doesn't setup forwards when --watch=false.
+	ctx, cancel := context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	f.WaitForAllPodsReady(ctx, "app=oneup")
+
+	var observedEventNames []string
+	for _, c := range f.mss.ma.Counts {
+		observedEventNames = append(observedEventNames, c.Name)
+	}
+
+	assert.Equal(t, 0, len(f.mss.ma.Counts))
+}
+
+func TestOptDefault(t *testing.T) {
+	f := newAnalyticsFixture(t)
+	defer f.TearDown()
+
+	f.SetOpt(analytics.OptDefault)
+
+	f.TiltUp("oneup")
+
+	// ForwardPort will fail if all the pods are not ready.
+	//
+	// We can't use the normal Tilt-managed forwards here because
+	// Tilt doesn't setup forwards when --watch=false.
+	ctx, cancel := context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	f.WaitForAllPodsReady(ctx, "app=oneup")
+
+	assert.Equal(t, 1, len(f.mss.ma.Counts))
```

Comment:
all the below should be wrapped in an `if assert.Equal(t, 1, len...)` or else `f.mss.ma.Counts[0]` will throw an index error if something went wrong and it'll be confusing.